sklearn/ensemble/tests/test_iforest.py

```diff
@@ -58,9 +58,9 @@ def test_iforest():
             IsolationForest(random_state=rng, **params).fit(X_train).predict(X_test)
```

Comment:
I assume that we also want the global random state here as well.

Reply:
I think that we should remove the line 30

```python
rng = np.random.RandomState(0)
```

We should not share the same random state in the tests. Instead we should locally create the random state internally in the function or instead use the global random state.

Reply:
I've updated the tests so that the global random state is now only used to permute the dataset.